crates/net/downloaders/src/bodies/bodies.rs

```diff
@@ -295,15 +295,16 @@ where
         }
 
         // Check if the provided range is the next expected range.
-        let is_next_range = range.start >= self.download_range.end;
-        if is_next_range {
+        let is_next_consecutive_range = range.start == self.download_range.end;
+        if is_next_consecutive_range {
```

Comment:
right, this makes sense,

this ensures we clear if we get a higher block from the CL